Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.h.in

```diff
@@ -17,17 +17,29 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#if PLATFORM(GTK)
 #if !defined(__WEBKIT2_H_INSIDE__) && !defined(WEBKIT2_COMPILATION)
 #error "Only <webkit2/webkit2.h> can be included directly."
 #endif
+#elif PLATFORM(WPE)
+#if !defined(__WEBKIT_H_INSIDE__) && !defined(WEBKIT2_COMPILATION)
+#error "Only <wpe/webkit.h> can be included directly."
+#endif
+#endif
 
 #ifndef WebKitAuthenticationRequest_h
 #define WebKitAuthenticationRequest_h
 
+#if PLATFORM(GTK)
 #include <gtk/gtk.h>
```

Comment:
And unifdef would still be needed for these cases where we really need a platform specific include.